Core/include/Acts/TrackFinding/CombinatorialKalmanFilter.hpp

```diff
@@ -309,190 +186,13 @@ class CombinatorialKalmanFilter {
-        TrackStateContainerBackend& trackStates, bool isOutlier,
-        const Logger& logger) const {
-      using PM = TrackStatePropMask;
-
-      using ResultTrackStateList =
-          Acts::Result<CkfTypes::BranchVector<TrackIndexType>>;
-      ResultTrackStateList resultTrackStateList{
-          CkfTypes::BranchVector<TrackIndexType>()};
-      CkfTypes::BranchVector<TrackIndexType>& trackStateList =
-          *resultTrackStateList;
-      trackStateList.reserve(end - begin);
-
-      std::optional<TrackStateProxy> firstTrackState{std::nullopt};
-      for (auto it = begin; it != end; ++it) {
-        auto& candidateTrackState = *it;
-
-        PM mask = PM::Predicted | PM::Filtered | PM::Jacobian | PM::Calibrated;
-        if (it != begin) {
-          // subsequent track states don't need storage for these as they will
-          // be shared
-          mask &= ~PM::Predicted & ~PM::Jacobian;
-        }
-        if (isOutlier) {
-          // outlier won't have separate filtered parameters
-          mask &= ~PM::Filtered;
-        }
-
-        // copy this trackstate into fitted states MultiTrajectory
-        auto trackState =
-            trackStates.makeTrackState(mask, candidateTrackState.previous());
-        ACTS_VERBOSE("Create SourceLink output track state #"
-                     << trackState.index() << " with mask: " << mask);
-
-        if (it != begin) {
-          // assign indices pointing to first track state
-          trackState.shareFrom(*firstTrackState, PM::Predicted);
-          trackState.shareFrom(*firstTrackState, PM::Jacobian);
-        } else {
-          firstTrackState = trackState;
-        }
-
-        // either copy ALL or everything except for predicted and jacobian
-        trackState.copyFrom(candidateTrackState, mask, false);
-
-        auto typeFlags = trackState.typeFlags();
-        typeFlags.set(TrackStateFlag::ParameterFlag);
-        typeFlags.set(TrackStateFlag::MeasurementFlag);
-        if (trackState.referenceSurface().surfaceMaterial() != nullptr) {
-          typeFlags.set(TrackStateFlag::MaterialFlag);
-        }
-        if (isOutlier) {
-          // propagate information that this is an outlier state
-          ACTS_VERBOSE(
-              "Creating outlier track state with tip = " << trackState.index());
-          typeFlags.set(TrackStateFlag::OutlierFlag);
-        }
-
-        trackStateList.push_back(trackState.index());
-      }
-      return resultTrackStateList;
-    }
-  };
-
   /// @brief Propagator Actor plugin for the CombinatorialKalmanFilter
   ///
-  /// @tparam source_link_accessor_t The type of source link accessor
   /// @tparam parameters_t The type of parameters used for "local" parameters.
   ///
   /// The CombinatorialKalmanFilter Actor does not rely on the measurements to
   /// be sorted along the track.
-  template <typename source_link_accessor_t, typename parameters_t>
+  template <typename parameters_t>
```

Comment:
I wonder if we could drop that? But that is something for another PR

Reply:
I actually wonder whether this can be different from BoundTrackParameters

Reply:
Right I think this can only be `BoundTrackParameters` since it should not be coupled to the start parameters. Even for the start parameters I would argue that only `BoundTrackParameters` make sense for now

Reply:
Something for a different PR